presto-hive/src/main/java/com/facebook/presto/hive/HivePartition.java

```diff
@@ -50,7 +55,19 @@ public HivePartition(SchemaTableName tableName, TupleDomain<HiveColumnHandle> ef
         this(tableName, effectivePredicate, UNPARTITIONED_ID, ImmutableMap.of(), bucket);
     }
 
-    public HivePartition(SchemaTableName tableName,
+    @JsonCreator
+    public HivePartition(
+            @JsonProperty("tableName") SchemaTableName tableName,
+            @JsonProperty("effectivePredicate") TupleDomain<HiveColumnHandle> effectivePredicate,
+            @JsonProperty("partitionId") String partitionId,
+            @JsonProperty("keysEntries") JsonSerializableMapEntries<ColumnHandle, SerializableNativeValue> keysEntries,
```

Comment:
Do you need the `JsonSerializableMapEntries` class? It seems like this will work if you just replace it with `List<Entry<ColumnHandle, SerializableNativeValue>>`